File: src/main/kotlin/dev/loveeev/astraTowny/events/nation/NationTownJoin.java
package dev.loveeev.astraTowny.events.nation;

import dev.loveeev.astratowny.objects.Town;
import lombok.Getter;
import org.bukkit.event.Cancellable;
import org.bukkit.event.Event;
import org.bukkit.event.HandlerList;
import org.jetbrains.annotations.NotNull;

public class NationTownJoin extends Event implements Cancellable {
    private static final HandlerList handlers = new HandlerList();
    private boolean isCancelled;
    @Getter
    private final Town town;

    public NationTownJoin(@NotNull Town town) {
        this.town = town;
        this.isCancelled = false; // Default to not cancelled
    }

    @Override
    public @NotNull HandlerList getHandlers() {
        return handlers;
    }

    public static HandlerList getHandlerList() {
        return handlers;
    }

    @Override
    public boolean isCancelled() {
        return isCancelled;
    }

    @Override
    public void setCancelled(boolean cancel) {
        this.isCancelled = cancel;
    }
}

File: src/main/kotlin/dev/loveeev/astraTowny/events/nation/NationTownJoin.kt
package dev.loveeev.astratowny.events.nation

import dev.loveeev.astratowny.objects.Town
import lombok.Getter
import org.bukkit.event.Cancellable
import org.bukkit.event.Event
import org.bukkit.event.HandlerList

class NationTownJoin(@field:Getter private val town: Town) : Event(), Cancellable {
    private var isCancelled = false

    override fun getHandlers(): HandlerList {
        return handlerList
    }

    override fun isCancelled(): Boolean {
        return isCancelled
    }

    override fun setCancelled(cancel: Boolean) {
        this.isCancelled = cancel
    }

    companion object {
        val handlerList: HandlerList = HandlerList()
    }
}